core/embed/rust/src/ui/component/base.rs

```diff
@@ -85,6 +85,10 @@ impl<T> Child<T> {
         self.component
     }
 
+    pub fn inner_mut(&mut self) -> &mut T {
+        &mut self.component
+    }
```

Comment:
Seems like `mutate` cannot be used everywhere - we need access to `ctx`, and we do not have it e.g. in `place()` - when I want to call `Child<Scrollbar>::set_page_count`, I would still need to use `inner_mut`. WIll change it to `mutate` in all places where `ctx` is available

Reply:
It's kind of the whole point of `Child` that whatever is inside can only be changed through `mutate` ...
Suggestion: in the `place()` method construct a new ScrollBar, set page count, then wrap it in Child and assign to `self.scrollbar`.

Reply:
Right, I haven't run into these because Model T page doesn't use `Child<T>` which might bite us once we have animations on a page.

> Either the whole Paginate::change_page would need to accept ctx

To me this sounds preferable to adding `inner_mut` to `Child`.

> Another problematic part would be getting page_count of the Child::inner, because that requires mutable access

Would forwarding the `Paginate` trait help here? If you also `impl Paginate for ScrollBar` this might even help with the first issue if I'm not missing something.

```rust
impl<T: Paginate> Paginate for Child<T>
```

Reply:
> To me this sounds preferable to adding inner_mut to Child.

It is being called from many places without access to `ctx`

> Would forwarding the Paginate trait help here? If you also impl Paginate for ScrollBar this might even help with the first issue if I'm not missing something.

Not sure how to do it - and how it would help with the mutable access and missing `ctx`

---

If we really want to remove `Child::inner_mut`, I cannot think of anything else than removing the `Child` wrapper around `Scrollbar` (and maybe some other places)